src/app/SecurityPanel/StoreJmxCredentials.tsx

```diff
@@ -94,7 +94,7 @@ const Component = () => {
   }, [context, context.notificationChannel, targets, setStoredTargets]);
```

Comment:
> Judging from your mock on line 94, it seems the NotificationChannel.messages subscription is invoked three times in StoreJmxCredentials. For my own understanding, could you explain why this happens? I thought there would be only two invocations (one for a stored notification and the other for a deleted notification) once the render is complete and the useEffect calls are triggered.

@hareetd, the list of discoverable `targets` is a dependency in the `useEffect` hook that subscribes to `TargetCredentialsStored` notifications, which causes this hook to be re-run when `targets` gets populated from another `useEffect` hook during render. This hook checks that the target parsed from the notification still exists in `targets` before adding it to the table.

Reply:
Makes sense, thanks!